yarn-project/circuits.js/src/abis/abis.ts

```diff
@@ -237,18 +240,27 @@ export function siloCommitment(wasm: IWasmModule, contract: AztecAddress, commit
  */
 export function computeVarArgsHash(wasm: IWasmModule, args: Fr[]): Promise<Fr> {
   if (args.length === 0) return Promise.resolve(Fr.ZERO);
-  if (args.length > 32 ** 2) throw new Error(`Cannot hash more than 1024 arguments`);
+  if (args.length > ARGS_HASH_CHUNK_SIZE * ARGS_HASH_CHUNK_COUNT)
+    throw new Error(`Cannot hash more than ${ARGS_HASH_CHUNK_SIZE * ARGS_HASH_CHUNK_COUNT} arguments`);
   wasm.call('pedersen__init');
 
   const wasmComputeVarArgs = (args: Fr[]) =>
     Fr.fromBuffer(wasmSyncCall(wasm, 'abis__compute_var_args_hash', new Vector(args), 32));
 
-  if (args.length > 32) {
-    const chunksHashes = chunk(args, 32).map(c => wasmComputeVarArgs(c));
-    return Promise.resolve(wasmComputeVarArgs(chunksHashes));
-  } else {
-    return Promise.resolve(wasmComputeVarArgs(args));
+  let chunksHashes = chunk(args, ARGS_HASH_CHUNK_SIZE).map(c => {
+    if (c.length < ARGS_HASH_CHUNK_SIZE) {
+      // Pad with zeroes
+      c = c.concat(Array(ARGS_HASH_CHUNK_SIZE - c.length).fill(Fr.ZERO));
+    }
+    return wasmComputeVarArgs(c);
+  });
+
+  if (chunksHashes.length < ARGS_HASH_CHUNK_COUNT) {
+    // Pad with zeroes
+    chunksHashes = chunksHashes.concat(Array(ARGS_HASH_CHUNK_COUNT - chunksHashes.length).fill(Fr.ZERO));
   }
+
+  return Promise.resolve(wasmComputeVarArgs(chunksHashes));
```

Comment:
I'm not sure we even need a non-default generator because functions have well defined argument lengths, so It's my understanding that we don't need to domain-separate those, because you won't be able to craft arguments of the same length that give the same hash. Maybe @iAmMichaelConnor or @suyash67 can help us with this question